api.lua

```diff
@@ -79,19 +79,29 @@ end
 
 
 minetest.register_on_leaveplayer(function(player)
-	force_detach(player)
+	if player then
+		force_detach(player)
+	end
 end)
 
 minetest.register_on_shutdown(function()
-    local players = minetest.get_connected_players()
+	local playerc
+	local players = minetest.get_connected_players()
 	for i = 1,#players do
-		force_detach(players[i])
+		playerc = players[i]
+		if playerc then
+			force_detach(players[i])
+		end
+		playerc = nil
 	end
```

Comment:
Although here, this would be a little better done as below, for semantic consistency. Not a big issue, can be done in a future cleanup.

```lua
    for i = 1,#players do
        playerc = players[i]

        if playerc then
            force_detach(playerc)
        end
    end
```

Reply:
yes noted.. but used the `players[i]` due i work with childrens that are not coders.. that's was the reason.. in fact if you noted many of my changes are focused in such things, thanks to pay attention @taikedz  i will review the rest of issues